### PR TITLE
ADR 008 Updating to tendermint v0.35.4

### PR DESCRIPTION
**Pull request by evan-forbes**
## Description

I decided to try to create a new ADR to go with #737 instead of modifying ADR 007, as there were quite a few modifications, and I felt like ADR 007 was already difficult to grok for someone outside of our org. This ADR organizes the commits in #737 and adds a sentence or two to each category to further explain the changes made. This PR also modifies ADR 007 to point towards ADR 008 as a more up to date reference. 

We don't have to go with this solution, we could instead modify ADR 007. I'm more than happy to do so if that's what we decide :slightly_smiling_face: 

ps: the rest of the ADRs are being added in #737

### CODE REVIEW

**Review by liamsi — APPROVED** [review]
👏🏼

File: docs/celestia-architecture/adr-007-minimal-changes-to-tendermint.md
Comment (adlerjohn) [inline docs/celestia-architecture/adr-007-minimal-changes-to-tendermint.md:6]: 🇺🇸
```suggestion
- 2021-08-20: Initial Description
- 2022-05-03: Update pointing to ADR 008
```

File: docs/celestia-architecture/adr-008-updating-to-tendermint-v0.35.x.md
Comment (adlerjohn) [inline docs/celestia-architecture/adr-008-updating-to-tendermint-v0.35.x.md:5]:
```suggestion
- 2022-05-03: Initial document describing changes to tendermint v0.35.x
```

**Review by adlerjohn — APPROVED** [review]

File: docs/celestia-architecture/adr-008-updating-to-tendermint-v0.35.x.md
Comment (evan-forbes) [reply docs/celestia-architecture/adr-008-updating-to-tendermint-v0.35.x.md:5]: damn, everytime lol

**Comment by evan-forbes** [comment]
will wait to merge this until after we merge #737 just in case something changes

**Review by SweeXordious — APPROVED** [review]

**Pull request merged**
